awscli/testutils.py

```diff
@@ -810,12 +811,12 @@ def get_key_contents(self, bucket_name, key_name):
         response = client.get_object(Bucket=bucket_name, Key=key_name)
         return response['Body'].read().decode('utf-8')
 
-    def key_exists(self, bucket_name, key_name):
-        client = self.create_client_for_bucket(bucket_name)
+    def key_exists(self, bucket_name, key_name, min_successes=3):
```

Comment:
I'm a little hesitant to use the waiter for `key_exists()` because we use this to assert it does not exist as well, which would mean we would definitely have to wait 100 seconds for each time we are trying to assert the key does not exist.